Server/JokesOnYou.Web.Api/Services/UserService.cs

```diff
@@ -37,5 +44,40 @@ public async Task UpdateUser(UserUpdateDTO updateDTO)
         {
             throw new NotImplementedException();
         }
+
+        public async Task<UserReplyDTO> LoginUser(UserLoginDTO userLogin)
+        {
+            var user = await _userRepository.GetUserByEmail(userLogin.Email);
+            if (user != null)
+            {
+                var signInResult = await _signInManager.CheckPasswordSignInAsync(user, userLogin.Password, false);
+
+                if (signInResult.Succeeded)
+                {
+                    var userReplyDTO = new UserReplyDTO()
+                    {
+                        Id = user.Id,
+                        Email = user.Email,
+                        UserName = user.Email,
+                        Role = "",
+                        Token = _tokenService.GetToken(user)
+                    };
+                    return userReplyDTO;
+                }
+                else
+                {
+                    throw new NotImplementedException("Sign in failed");
+                }
+            }
+            else
+            {
+                throw new NotImplementedException("User not found"); 
```

Comment:
Return Unautorized instead?

Reply:
Service should not return any http, so throwing the exception is good but I would change it to AppException because it's not not implemented. The global exception handler will handle the return.

Reply:
> Service should not return any http, so throwing the exception is good but I would change it to AppException because it's not not implemented. The global exception handler will handle the return.

Use the built in `throw new UnauthorizedAccessException();` then just add it to the ones being catched inside the global ex handler and return a 401 unauthorized